app/controllers/conversations_controller.rb

```diff
@@ -5,19 +5,26 @@ class ConversationsController < ApplicationController
 
   # GET /conversations or /conversations.json
   def index
+    redirect_back fallback_location: root_url, alert: "You're not authorized for that."
     @conversations = Conversation.all
   end
 
   # GET /conversations/1 or /conversations/1.json
-  def show; end
+  def show
+    if current_user != @conversation.customer and current_user != @conversation.company_representative
```

Comment:
I would avoid using `and` in conditionals and prefer `&&` since [they are very *slightly* different](https://stackoverflow.com/a/1426835).